runbotics/runbotics-scheduler/src/utils/decorators/user.decorator.ts

```diff
@@ -1,8 +1,12 @@
 import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import { AuthRequest } from '#/types';
+import { UserEntity } from '#/database/user/user.entity';
 
 export const User = createParamDecorator(
-    (_data: unknown, ctx: ExecutionContext) => {
-        const request = ctx.switchToHttp().getRequest();
-        return request.user;
+    (data: keyof UserEntity | undefined, ctx: ExecutionContext) => {
```

Comment:
you can pick a better name for data since the type is known now. 
unknown => data?: keyof UserEntity